crates/stages/src/stages/tx_lookup.rs

```diff
@@ -92,43 +96,60 @@ impl<DB: Database> Stage<DB> for TransactionLookupStage {
             return Ok(ExecOutput::done(input.checkpoint()))
         }
 
-        let (tx_range, block_range, is_final_range) =
-            input.next_block_range_with_transaction_threshold(provider, self.commit_threshold)?;
-        let end_block = *block_range.end();
+        // 500MB temporary files
+        let mut hash_collector: Collector<TxHash, TxNumber> =
+            Collector::new(Arc::new(TempDir::new()?), 500 * (1024 * 1024));
 
-        debug!(target: "sync::stages::transaction_lookup", ?tx_range, "Updating transaction lookup");
+        loop {
+            let (tx_range, block_range, is_final_range) = input
+                .next_block_range_with_transaction_threshold(provider, self.commit_threshold)?;
 
-        let mut tx_list = provider.transaction_hashes_by_range(tx_range)?;
+            let end_block = *block_range.end();
 
-        // Sort before inserting the reverse lookup for hash -> tx_id.
-        tx_list.par_sort_unstable_by(|txa, txb| txa.0.cmp(&txb.0));
+            debug!(target: "sync::stages::transaction_lookup", ?tx_range, "Updating transaction lookup");
```

Comment:
nit: move this below when we're actually inserting the data into the database, and add an ETL debug log here

Reply:
changed it, but not quite like this. we insert the whole range into the database, so i think it's better to provide intermediate %